src/compas_ui/scene.py

```diff
@@ -18,24 +19,26 @@ def clear_scene(self):
     raise NotImplementedError
 
 
-class Scene(object):
+class Scene(Data, Singleton):
 
-    def __init__(self, app, settings={}):
+    def __init__(self, app=None, settings={}):
         self.app = app
-        self.nodes = {}
+        self.objects = set()
         self.settings = settings
 
     def add(self, item, **kwargs):
         node = Object(item, **kwargs)
-        guid = uuid4()
-        self.nodes[guid] = node
+        guid = str(uuid4())
+        self.objects.add(guid)
+        self.app.session['objects'][guid] = node
```

Comment:
i think the scene should take care of this, since it should be usable outside of the context of an app...